pkg/storage/iterator.go

```diff
@@ -421,7 +431,25 @@ func fetchLazyChunks(ctx context.Context, chunks []*chunkenc.LazyChunk) error {
 			lastErr = err
 		}
 	}
-	return lastErr
+
+	if lastErr != nil {
+		return lastErr
+	}
+
+	for _, c := range chunks {
+		if c.Chunk.Data != nil {
+			c.IsValid = true
+		}
+	}
+	return nil
+}
+
+func isInvalidChunkError(err error) bool {
+	err = errors.Cause(err)
+	if err, ok := err.(promql.ErrStorage); ok {
+		return err.Err == chunk.ErrInvalidChecksum
```

Comment:
I thought the error could be this one and the Loki one ? shouldn't you test that ?

Reply:
Yeah, we can add the Loki one as well. It's just the same error. Will change it

Reply:
@cyriltovena I have made this change. I will rebase on top of this https://github.com/grafana/loki/pull/1914

Reply:
> I thought the error could be this one __and__ the Loki one ?

I think it can be both. You also need a small test for that function, this was the point.

Almost there !

Reply:
Should we just use `chunk.ErrInvalidChecksum` from cortex here as well https://github.com/grafana/loki/blob/master/pkg/chunkenc/memchunk.go#L209 ?

If any change on cortex, we should make sure to bring that change to Loki as well otherwise we would break

Reply:
@cyriltovena Added the test

Reply:
Series test is failing. Have to identify some other way to do chunk test. Doing this doesn't work out
https://github.com/grafana/loki/blob/7c2274093f7a76f433a6fd32aad2bd0205e7c1c6/pkg/storage/util_test.go#L208

I will re work on this

Reply:
Changed the Batchsize in `Test_store_GetSeries` test. That works.

WDYT we should?